src/test/java/de/tum/in/www1/artemis/RatingResourceIntegrationTest.java

```diff
@@ -0,0 +1,136 @@
+
+    @Autowired
+    ExerciseRepository exerciseRepo;
+
+    @Autowired
+    ResultRepository resultRepo;
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RatingService ratingService;
+
+    @Autowired
+    RatingRepository ratingRepo;
+
+    @Autowired
+    SubmissionRepository submissionRepo;
+
+    private TextExercise exercise;
+
+    private List<User> users;
+
+    private TextSubmission submission;
+
+    private Result result;
+
+    private Rating rating;
+
+    @BeforeEach
+    public void initTestCase() {
+        users = database.addUsers(2, 1, 1);
+        database.addCourseWithOneTextExercise();
+        exercise = (TextExercise) exerciseRepo.findAll().get(0);
+        User student1 = users.get(0);
+        Participation participation = database.addParticipationForExercise(exercise, student1.getLogin());
+
+        submission = ModelFactory.generateTextSubmission("example text", Language.ENGLISH, true);
+        submission = database.addTextSubmission(exercise, submission, student1.getLogin());
+
+        result = ModelFactory.generateResult(true, 0);
+        result = database.addResultToSubmission(submission);
+
+        rating = new Rating();
+        rating.setResult(result);
+        rating.setRating(2);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        ratingRepo.deleteAll();
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testCreateRating_asUser() throws Exception {
+        request.post("/api/results/" + result.getId() + "/rating/" + rating.getRating(), null, HttpStatus.CREATED);
+    }
+
+    @Test
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void testCreateRating_asTutor_FORBIDDEN() throws Exception {
+        request.post("/api/results/" + result.getId() + "/rating/" + rating.getRating(), null, HttpStatus.FORBIDDEN);
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testGetRating_asUser() throws Exception {
+        Rating result = ratingService.saveRating(this.result.getId(), rating.getRating());
+        request.get("/api/results/" + result.getId() + "/rating", HttpStatus.OK, Rating.class);
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testGetRating_asUser_NOT_FOUND() throws Exception {
+        request.get("/api/results/" + result.getId() + "/rating", HttpStatus.NOT_FOUND, Rating.class);
+    }
+
+    @Test
+    @WithMockUser(value = "student1", roles = "USER")
+    public void testUpdateRating_asUser() throws Exception {
+        Rating result = ratingService.saveRating(this.result.getId(), rating.getRating());
+        request.put("/api/results/" + result.getId() + "/rating/" + rating.getRating(), null, HttpStatus.OK);
```

Comment:
Assertions are missing. Also, what is being tested here? Wouldn't it be better if an actual change to the rating value was tested and verified?